### PR TITLE
Updating blockblock from 0.9.9.4 to 1.0.1 for MacOS >= Catalina

### PR DESCRIPTION
**Pull request by benmontour-wf**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

This is an attempted fix for #81252. I'm not sure if putting if/else stanzas where I have will work, but I'm open to suggestions.

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask

### CODE REVIEW

File: Casks/blockblock.rb [hunk 1]
```diff
@@ -1,24 +1,48 @@
...
+                        args:       ['-install'],
+                        sudo:       true,
+                      }
+
+    uninstall script: {
+                        executable: "#{staged_path}/BlockBlock Installer.app/Contents/MacOS/BlockBlock Installer",
+                        args:       ['-uninstall'],
+                        sudo:       true,
+                      }
+  end
```
Comment (vitorgalvao) [inline Casks/blockblock.rb:46]: Needs a new line below.

File: Casks/blockblock.rb [hunk 2]
```diff
@@ -1,24 +1,48 @@
...
+    version '1.0.1'
+    sha256 '7e5c459ce033eda44a6e5e2e57ec503ad07d96b6272d0ee7c3bbfb6a04d03ce4'
 
+    # bitbucket.org/objective-see/ was verified as official when first introduced to the cask
+    url "https://bitbucket.org/objective-see/deploy/downloads/BlockBlock_#{version}.zip"
+    appcast 'https://objective-see.com/products/changelogs/BlockBlock.txt'
+    name 'BlockBlock'
+    homepage 'https://objective-see.com/products/blockblock.html'
+
+    depends_on macos: '>= :catalina'
```
Comment (vitorgalvao) [inline Casks/blockblock.rb:33]: Redundant. Please remove.

File: Casks/blockblock.rb [hunk 3]
```diff
@@ -1,24 +1,48 @@
...
+    sha256 '6ab3a8224e8bc77b9abe8d41492c161454c6b0266e60e61b06931fed4b431282'
 
-  # bitbucket.org/objective-see/ was verified as official when first introduced to the cask
-  url "https://bitbucket.org/objective-see/deploy/downloads/BlockBlock_#{version}.zip"
-  appcast 'https://objective-see.com/products/changelogs/BlockBlock.txt'
-  name 'BlockBlock'
-  homepage 'https://objective-see.com/products/blockblock.html'
+    # bitbucket.org/objective-see/ was verified as official when first introduced to the cask
+    url "https://bitbucket.org/objective-see/deploy/downloads/BlockBlock_#{version}.zip"
+    appcast 'https://objective-see.com/products/changelogs/BlockBlock.txt'
```
Comment (vitorgalvao) [inline Casks/blockblock.rb:8]: Remove this one. The appcast belongs only to the most recent.

File: Casks/blockblock.rb [hunk 4]
```diff
@@ -1,24 +1,48 @@
...
+                        args:       ['-uninstall'],
+                        sudo:       true,
+                      }
+  else
+    version '1.0.1'
+    sha256 '7e5c459ce033eda44a6e5e2e57ec503ad07d96b6272d0ee7c3bbfb6a04d03ce4'
 
+    # bitbucket.org/objective-see/ was verified as official when first introduced to the cask
+    url "https://bitbucket.org/objective-see/deploy/downloads/BlockBlock_#{version}.zip"
+    appcast 'https://objective-see.com/products/changelogs/BlockBlock.txt'
```
Comment (vitorgalvao) [inline Casks/blockblock.rb:29]: `appcast`, `name`, and `homepage` need to be outside the conditional.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

File: Casks/blockblock.rb (hunk 4 above)
Comment (benmontour-wf) [reply Casks/blockblock.rb]: moved to top of file, outside conditional

File: Casks/blockblock.rb (hunk 3 above)
Comment (benmontour-wf) [reply Casks/blockblock.rb]: appcast was moved to top of file, outside conditional

File: Casks/blockblock.rb (hunk 2 above)
Comment (benmontour-wf) [reply Casks/blockblock.rb]: removed

File: Casks/blockblock.rb (hunk 1 above)
Comment (benmontour-wf) [reply Casks/blockblock.rb:42]: added newline

**Review by benmontour-wf — COMMENTED** [review]
updated based on suggested changed

File: Casks/blockblock.rb [hunk 5]
```diff
@@ -1,24 +1,45 @@
 cask 'blockblock' do
-  version '0.9.9.4'
-  sha256 '6ab3a8224e8bc77b9abe8d41492c161454c6b0266e60e61b06931fed4b431282'
-
-  # bitbucket.org/objective-see/ was verified as official when first introduced to the cask
-  url "https://bitbucket.org/objective-see/deploy/downloads/BlockBlock_#{version}.zip"
   appcast 'https://objective-see.com/products/changelogs/BlockBlock.txt'
```
Comment (vitorgalvao) [inline Casks/blockblock.rb:2]: These should be below the conditional, for consistency with other casks.

Comment (benmontour-wf) [reply Casks/blockblock.rb]: Moved appcast, name, and homepage to bottom of file for consistency

**Pull request merged**